x-pack/elastic-agent/pkg/agent/application/pipeline/dispatcher/dispatcher.go

```diff
@@ -87,18 +95,18 @@ func (ad *ActionDispatcher) Dispatch(acker store.FleetAcker, actions ...fleetapi
 	)
 
 	for _, action := range actions {
-		if err := ad.ctx.Err(); err != nil {
+		if err = ad.ctx.Err(); err != nil {
 			return err
 		}
 
-		if err := ad.dispatchAction(action, acker); err != nil {
+		if err = ad.dispatchAction(action, acker); err != nil {
 			ad.log.Debugf("Failed to dispatch action '%+v', error: %+v", action, err)
 			return err
 		}
 		ad.log.Debugf("Successfully dispatched action: '%+v'", action)
 	}
 
-	return acker.Commit(ad.ctx)
+	return acker.Commit(ctx)
```

Comment:
@AndersonQ @ph I just noticed this while looking through the diff. Is this change safe?

Reply:
We're on an odd scenario here, there are 2 contexts for this method, this shouldn't happen.
 - `ad.ctx` is for cancelation, it shuts the `ActionDispatcher` down
 - `ctx` is just a carrier for the span.
 
Any proper solution would require a good amount of refactoring. Also passing in just the span isn't ideal.

For simplicity, I suggest to document that, let's comment the method and make clear that for now the only allowed function for the context received as parameter is to carry the span, and that cancelation is done through the `ActionDispatcher` context.

I'm adding my suggestion to fix that

@ph, what do you think?